packages/ember-views/lib/views/view.js

```diff
@@ -1022,6 +1022,8 @@ var View = CoreView.extend({
     @return Ember.View
   */
   nearestChildOf: function(klass) {
+    Ember.deprecate("nearestChildOf has been deprecated.\nFor existing behavior, use nearestOfType.");
```

Comment:
Technically, it's not in favor. I'd like to make it clear to people who try to call this (like myself) that this doesn't do what is expected while marking a clear deprecation path for those who are currently using this.